src/main/java/ArraySortedList.java

```diff
@@ -0,0 +1,184 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ArraySortedList<T> implements SortedList<T> {
+
+    private static final int DEFAULT_CAPACITY = 100;
+    protected T[] list;
+    protected int rear;
+
+    public ArraySortedList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public ArraySortedList(int initialCapacity) {
+        list = (T[]) new Object[initialCapacity];
+        rear = 0;
+    }
+
+    /**
+     * Doubles the size of the list array and copy the
+     * contents over.
+     */
+    @SuppressWarnings("unchecked")
+    private void expandCapacity() {
+        T[] newList = (T[]) new Object[list.length * 2];
+        for (int i = 0; i < list.length; ++i) {
+            newList[i] = list[i];
+        }
+        list = newList;
+    }
+
+    /**
+     * Shifts elements in an array down (towards index 0) towards
+     * the starting index specified. This method assumes that
+     * the calling function manages the rear field.
+     *
+     * @param start Index of where shift starts/stops and element
+     *              overwritten.
+     */
+    private void shiftLeft(int start) {
+        for (int i = start; i < list.length - 1; ++i) {
+            list[i] = list[i + 1];
+        }
+        list[list.length - 1] = null;
+    }
+
+    /**
+     * Shifts elements in an array up (towards index rear) away
+     * from the starting index specified. This method assumes that
+     * the calling function manages the rear field and any
+     * need of expandCapacity().
+     *
+     * @param start Index of where shifting starts/stops and where
+     *              space is opened.
+     */
+    private void shiftRight(int start) {
+        for (int i = rear; i > start; --i) {
+            list[i] = list[i - 1];
+        }
+        list[start] = null;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void add(T element) {
+        if (size() == list.length) {
+            expandCapacity();
+        }
+        // Assume the element has some defined ordering
+        Comparable<T> comp = (Comparable<T>) element;
```

Comment:
A little spooky. You could enforce this with more type inheritance,

```java
public class SortedList<T extends Comparable<? super T>> implements List<T> { }
```

`Comparable<? super T>` is technically required (as opposed to `Comparable<T>`). It means that subclasses are also captured here, for example, let's assume `Number` was `Comparable`.

```java
interface Number implements Comparable<Number> { }

class Integer implements Number {
    @Override
    public int compareTo(Number num) { /* whatevs */ }
}
```
`Integer` there is a `Comparable<Number>`. If you wrote the list as `T extends Comparable<T>`, then `Integer` would not conform, since generics are invariant. `? super T` means any supertype of `Integer`, meaning `Number` -- so it'd conform again. `? super T` can also be `T`.